git_repo/services/ext/github.py

```diff
@@ -13,15 +13,24 @@
 
 from datetime import datetime
 
+GITHUB_COM_FQDN = 'github.com'
+
 @register_target('hub', 'github')
 class GithubService(RepositoryService):
-    fqdn = 'github.com'
+    fqdn = GITHUB_COM_FQDN
 
     def __init__(self, *args, **kwarg):
         self.gh = github3.GitHub()
         super(GithubService, self).__init__(*args, **kwarg)
 
     def connect(self):
+        if self.fqdn != GITHUB_COM_FQDN:
+            # upgrade self.gh from a GitHub object to a GitHubEnterprise object
+            gh = github3.GitHubEnterprise(RepositoryService.build_url(self))
+            self.gh._session.base_url = gh._session.base_url
```

Comment:
OK I fixed it.

- self.gh must change to new one(GitHubEnterprise)
- self.gh._session must keeps old one(created by __init__)
- self.gh._session.base_url must change to new URL

```python
            gh = github3.GitHubEnterprise(RepositoryService.build_url(self))
            self.gh._session.base_url = gh._session.base_url
            gh._session = self.gh._session
            self.gh = gh
```